### ISSUE TITLE
[Question] Could you please provide a syntax example for multi item phabricator-lint file?

### ISSUE THREAD

**Issue by OCram85** [uber/phabricator-jenkins-plugin#243]
Hi,
I followed your advanced guide and tried to build a phabricator-lint file for multiple src file:

These is the syntax I tried:
```YAML
{ 
    {
        "name":  "PSAvoidUsingCmdletAliases",
        "code":  "PSAvoidUsingCmdletAliases",
        "severity":  "warning",
        "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
        "line":  139,
        "char":  50,
        "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
    },
    {
        "name":  "PSAvoidUsingCmdletAliases",
        "code":  "PSAvoidUsingCmdletAliases",
        "severity":  "warning",
        "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
        "line":  145,
        "char":  46,
        "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
    }
}
```

```YAML
[ 
    {
        "name":  "PSAvoidUsingCmdletAliases",
        "code":  "PSAvoidUsingCmdletAliases",
        "severity":  "warning",
        "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
        "line":  139,
        "char":  50,
        "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
    },
    {
        "name":  "PSAvoidUsingCmdletAliases",
        "code":  "PSAvoidUsingCmdletAliases",
        "severity":  "warning",
        "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
        "line":  145,
        "char":  46,
        "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
    }
]
```

```YAML
{
    [ 
        {
            "name":  "PSAvoidUsingCmdletAliases",
            "code":  "PSAvoidUsingCmdletAliases",
            "severity":  "warning",
            "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
            "line":  139,
            "char":  50,
            "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
        },
        {
            "name":  "PSAvoidUsingCmdletAliases",
            "code":  "PSAvoidUsingCmdletAliases",
            "severity":  "warning",
            "path":  "C:\\Jenkins\\workspace\\VMWARE\\src\\Backup\\Backup-VCSAToFile.ps1",
            "line":  145,
            "char":  46,
            "description":  "\u0027Select\u0027 is an alias of \u0027Select-Object\u0027. Alias can introduce possible problems and make scripts hard to maintain. Please consider changing alias to its full content."
        }
    ]
}
```

But unfortunately I alway run into this error:
```
Archiving artifacts
[phabricator:plugin-provider] 'cobertura' plugin not installed.
[phabricator:process-build-result] No unit results available.
[phabricator:process-build-result] No coverage provider available.
net.sf.json.JSONException: A JSONObject text must begin with '{' at character 1 of ï»¿{
	at net.sf.json.util.JSONTokener.syntaxError(JSONTokener.java:499)
	at net.sf.json.JSONObject._fromJSONTokener(JSONObject.java:919)
	at net.sf.json.JSONObject._fromString(JSONObject.java:1145)
	at net.sf.json.JSONObject.fromObject(JSONObject.java:162)
	at net.sf.json.JSONObject.fromObject(JSONObject.java:132)
	at com.uber.jenkins.phabricator.BuildResultProcessor.processLintResults(BuildResultProcessor.java:153)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.perform(PhabricatorNotifier.java:203)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:735)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:676)
	at hudson.model.Build$BuildExecution.post2(Build.java:186)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:621)
	at hudson.model.Run.execute(Run.java:1760)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:405)
```

**Comment by OCram85** [comment]
Do you have an idea what's going wrong here? - I'm not quite sure if its the syntax or the content.
The Jenkins build slave is running on a Windows Server OS.
Could the error `A JSONObject text must begin with '{' at character 1 of ï»¿{` be caused by an strange file encoding eg. utf8-bom ?

**Comment by artms** [comment]
Failure you see is not caused by plugin itself but by library which is responsible for parsing json. It looks like your script/job is generating extra symbols at the beginning of lint file, in worst case try sed'ing out those symbols.

Format of phabricator-lint was provided in https://github.com/uber/phabricator-jenkins-plugin/issues/162#issuecomment-228397768

**Comment by OCram85** [comment]
You`re right. There was a problem with the file encoding. The file is created by PowerShell and the default encoding was `UTF8 with BOM`. I removed the BOM but now I get the next error:

`net.sf.json.JSONException: Found starting '[' but missing ']' at the end. at character 0 of null`

**Comment by artms** [comment]
Did you check: https://github.com/uber/phabricator-jenkins-plugin/issues/162#issuecomment-228397768 - make sure json file is valid json, for example run `json_verify` command or anything else similar available on your platform. net.sf.json.JSONException is json parsing library exception which is not part of plugin and is industry stardard so most likely json is incorrect

**Comment by OCram85** [comment]
I checked the output against a JSON validation tool. The `.arclint` contains a compressed json in one line.

**Comment by anonfriese** [comment]
According to https://github.com/uber/phabricator-jenkins-plugin/blob/master/docs/advanced.md#custom-lint the lintfile shouldn't actually contain a valid JSON List/Dict but instead every Line should consist of a single JSON Dict in itself. This means every Line must be a valid JSON-Dict with the needed Keys, but the file itself won't be valid JSON.

With that knowledge I got so far as to get 5 Lines parsed and shown up in Phabricator. Unfortunately on line no. 6 it fails with `net.sf.json.JSONException: Found starting '{' but missing '}' at the end. at character 0 of null` and i haven't got a clue as to why. The Lint-Result that seems to fail is indeed the same as the previous line, same description, code, file etc., only a different Line :-(
